common/grammar-parser.cpp

```diff
@@ -137,6 +152,86 @@ namespace grammar_parser {
             bool                                 is_nested) {
         size_t last_sym_start = out_elements.size();
         const char * pos = src;
+
+        auto handle_repetitions = [&](size_t min_times, int max_times) {
```

Comment:
Might have trouble mixing size_t (unsigned) and int (signed) types in comparisons -- maybe make both `int`?

Reply:
Ah thanks, fixed